### PR TITLE
Teacher dashboard/ rearrange classes

### PR DESCRIPTION
**Pull request by eadams17**
## WHAT
add the ability to rearrange teacher classes on the dashboard

## WHY
teachers have requested this feature to be able to better organize their classes

## HOW
move the teacher cards into a sortable list and store the classroom sorted order in `classrooms_teachers`

### Screenshots
(If applicable. Also, please censor any sensitive data)
Screen Shot 2021-07-01 at 4 57 38 PM

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)
https://www.notion.so/quill/Allow-teachers-to-re-arrange-classes-642d0f6b976d486db6df1aaa34d89ee6

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  Yes
Have you deployed to Staging? | Yes
Self-Review: Have you done an initial self-review of the code below on Github? | Yes
Design Review: If applicable, have you compared the coded design to the mockups? | Yes

### CODE REVIEW

**Review by anathomical — APPROVED** [review]
This looks fine, but are we okay with this order only being in localStorage?  Did we talk to Peter S about that just so he knows that's how we're implementing it?  (Probably fine, just don't want anyone to be surprised.)

**Review by emilia-friedberg — APPROVED** [review]

File: services/QuillLMS/spec/controllers/classrooms_teachers_controller_spec.rb [hunk 1]
```diff
@@ -46,6 +46,28 @@
   #   end
   # end
 
+  describe '#update_order' do
```
Comment (anathomical) [inline services/QuillLMS/spec/controllers/classrooms_teachers_controller_spec.rb:49]: Nice.  Love the tests.

File: services/QuillLMS/app/controllers/teachers/classrooms_controller.rb [hunk 2]
```diff
@@ -173,7 +173,9 @@ def transfer_ownership
   end
 
   private def format_classrooms_for_index
-    classrooms = Classroom.unscoped.order(created_at: :desc).joins(:classrooms_teachers).where(classrooms_teachers: {user_id: current_user.id})
+    has_classroom_order = ClassroomsTeacher.where(user_id: current_user.id).all? { |classroom| classroom.order }
+    classrooms = Classroom.unscoped.joins(:classrooms_teachers).where(classrooms_teachers: {user_id: current_user.id})
+    classrooms = has_classroom_order ? classrooms.includes(:classrooms_teachers).order('classrooms_teachers.order') : classrooms.order(created_at: :desc)
```
Comment (anathomical) [inline services/QuillLMS/app/controllers/teachers/classrooms_controller.rb:178]: Might be worth adding a test case for this behavior, just to ensure that we don't break sorting later on somehow.  So a case where they all have order, and a case where only some have order, would probably cover us here.

**Review by anathomical — CHANGES_REQUESTED** [review]
Looks good.  I would like to see a couple more test cases around sorting, but all of the code looks good to me.

**Review by anathomical — APPROVED** [review]

**Pull request merged**